tools/build-tools/src/buildVersion/buildVersion.ts

```diff
@@ -90,6 +92,42 @@ export function getSimpleVersion(file_version: string, arg_build_num: string, ar
     return fullVersion;
 }
 
+type TagPrefix = string | undefined | "client" | "server";
+
+/**
+ * Extracts versions from a list of strings, sorts them according to semver rules, and returns the sorted array.
+ *
+ * @param prefix - the tag prefix to filter the tags by (client, server, etc.)
+ * @param input_tags - a list of tags. This function will execute `git tags -l` to get the list if this parameter is undefined.
+ * @returns an array of versions extracted from the provided tags
+ */
+export function getTaggedVersions(prefix: TagPrefix, input_tags?: string[]) {
+    const input = child_process.execSync(`git tag -l`, { encoding: "utf8" });
```

Comment:
This shouldn't be necessary  if input_tags is passed in.

Maybe you should split this into two functions?  One that doesn't receive `input_tags` and would exec `git tag -l`, and it will call the one that will take the input_tags with parameters?

Reply:
Good idea.

Reply:
Done in latest.